ingestion/src/metadata/ingestion/source/database/dbt.py

```diff
@@ -510,16 +510,18 @@ def create_dbt_query_lineage(
         logger.info(f"Processing DBT Query lineage for: {table_fqn}")
 
         try:
-            source_elements = table_fqn.split(".")
+            source_elements = fqn.split(table_fqn)
+            # remove service name from fqn to make it parseable in format db.schema.table
+            query_fqn = ".".join(source_elements[-3:])
```

Comment:
Instead of using the `".".join(...)` we should use the methods in `fqn.py`. IMO we could rename the `_build` to `join` or something and use it in these scenarios to have proper quoting and centralising the dot logic